sim/midas/src/main/scala/midas/widgets/PlusArgsBridge.scala

```diff
@@ -0,0 +1,210 @@
+
+  generateAnnotations()
+}
+
+object PlusArgsBridge {
+
+  /** Helper for creating the bridge. All parameter requirements are checked here
+    *
+    * @param clock
+    *   The clock
+    *
+    * @param reset
+    *   The reset
+    *
+    * @param params
+    *   Describes the name, width and default PlusArg value
+    */
+  private def annotatePlusArgsBridge(clock: Clock, reset: Reset, params: PlusArgsBridgeParams): PlusArgsBridge = {
+    val target = Module(new PlusArgsBridge(params))
+    target.io.clock := clock
+    target
+  }
+
+  /** apply overload which takes the parameters without the case class The signature was copied from
+    * chipyard/generators/rocket-chip/src/main/scala/util/PlusArg.scala
+    *
+    * @param name
+    *   string identifier, should include "name=%d"
+    *
+    * @param default
+    *   The value of the register if no matching PlusArg is provided
+    *
+    * @param docstring
+    *   The doctring
+    *
+    * @param width
+    *   The width of the register, in bits
+    */
+  // Signature copied from chipyard/generators/rocket-chip/src/main/scala/util/PlusArg.scala
+  def apply(name: String, default: BigInt = 0, docstring: String = "", width: Int = 32): PlusArgsBridge = {
+    val params = PlusArgsBridgeParams(name, default, docstring, width)
+    annotatePlusArgsBridge(Module.clock, Module.reset, params)
+  }
+
+  /** apply overload which takes the parameters case class
+    *
+    * @param params
+    *   Describes the name, width and default PlusArg value
+    */
+  def apply(params: PlusArgsBridgeParams): PlusArgsBridge = {
+    annotatePlusArgsBridge(Module.clock, Module.reset, params)
+  }
+
+  /** similar to apply, but named drive. This overload returns the out value. This overload is probably the one you want
+    *
+    * @param params
+    *   Describes the name, width and default PlusArg value
+    */
+  def drive(params: PlusArgsBridgeParams): UInt = {
+    val target = annotatePlusArgsBridge(Module.clock, Module.reset, params)
+    target.io.out
+  }
+}
+
+/** The host-side implementation. Calculates widths and bundles multiple 32 bit MMIO to the exact width of
+  * `params.width`. If `params.width` does not evenly divide by 32 a MMIO with the remainder bits will be created. The
+  * remainders name will always be `outN` where `N` is the last MMIO
+  *
+  * `initDone` is used as valid for the outChannel. The C++ driver ensures that all `outN` are driven before `initDone`
+  * is asserted
+  *
+  * @param params
+  *   Describes the name, width and default PlusArg value
+  */
+class PlusArgsBridgeModule(params: PlusArgsBridgeParams)(implicit p: Parameters)
+    extends BridgeModule[PlusArgsBridgeHostIO]()(p) {
+  lazy val module = new BridgeModuleImp(this) {
+
+    val io    = IO(new WidgetIO())
+    val hPort = IO(new PlusArgsBridgeHostIO(params)())
+
+    // divide with a ceiling round, to get the total number of slices
+    val sliceCount = (params.width + ctrlWidth - 1) / ctrlWidth;
```

Comment:
```suggestion
    val sliceCount = (params.width + ctrlWidth - 1) / ctrlWidth
```

No semicolons in Scala unless multiple statements are needed in a single line.